src/us/kbase/workspace/database/provenance/CreditMetadata.java

```diff
@@ -0,0 +1,465 @@
+				&& Objects.equals(funding, other.funding)
+				&& Objects.equals(identifier, other.identifier)
+				&& Objects.equals(license, other.license)
+				&& Objects.equals(relatedIdentifiers, other.relatedIdentifiers)
+				&& Objects.equals(resourceType, other.resourceType)
+				&& Objects.equals(titles, other.titles)
+				&& Objects.equals(version, other.version);
+	}
+
+	/**
+	 * Gets a builder for the {@link CreditMetadata}.
+	 *
+	 * @param identifier
+	 *                unique persistent ID for the resource
+	 * @param contributors
+	 *                list of {@link Contributor} objects; at least one contributor is required
+	 * @param titles
+	 *                list of {@link Title} objects; at least one title is required
+	 * @return the builder.
+	 */
+	public static Builder getBuilder(
+			final String identifier,
+			final List<Contributor> contributors,
+			final List<Title> titles) {
+		return new Builder(identifier, contributors, titles);
+	}
+
+	/** A builder for {@link CreditMetadata}. */
+	public static class Builder {
+		private final String identifier;
+		private String license = null;
+		private String version = null;
+		private ResourceType resourceType = ResourceType.DATASET;
+		private List<String> comments = null;
+		private final List<Contributor> contributors;
+		private List<EventDate> dates = null;
+		private List<FundingReference> funding = null;
+		private List<PermanentID> relatedIdentifiers = null;
+		private final List<Title> titles;
+		private List<String> errorList = new ArrayList<>();
+
+		private Builder(
+				final String identifier,
+				final List<Contributor> contributors,
+				final List<Title> titles) {
+
+			String checkedPid = identifier;
+			try {
+				checkedPid = Common.checkPid(identifier, "identifier", false);
+			} catch (IllegalArgumentException e) {
+				this.errorList.add(e.getMessage());
+			}
+			this.identifier = checkedPid;
+
+			// must be at least one contributor
+			this.contributors = Common.dedupeSimpleList(contributors);
+			if (this.contributors == null || this.contributors.isEmpty()) {
+				this.errorList.add("at least one contributor must be provided");
+			}
+
+			// must be at least one title
+			this.titles = Common.dedupeSimpleList(titles);
+			if (this.titles == null || this.titles.isEmpty()) {
+				this.errorList.add("at least one title must be provided");
+			}
+		}
+
+		/**
+		 * Sets the resource type for the resource.
+		 *
+		 * As there is currently only one resource type, entering null will set the
+		 * resource type value to DEFAULT_RESOURCE_TYPE in the resultant object.
+		 *
+		 * @param resourceType
+		 *                resource type as a string
+		 * @return this builder
+		 */
+		public Builder withResourceTypeString(final String resourceType) {
+			if (resourceType != null) {
+				try {
+					this.resourceType = ResourceType.getResourceType(resourceType);
+				} catch (IllegalArgumentException e) {
+					// TEMPORARY HACK
```

Comment:
I don't think this behavior will change when there is more than one resource type, right? whitespace will still be an error and the message will still need editing, and null should still default to `DEFAULT_RESOURCE_TYPE`

Reply:
N/A

Reply:
:+1: